pennylane/interfaces/tf.py

```diff
@@ -99,30 +132,17 @@ def grad(grad_output, **tfkwargs):
             """Returns the vector-Jacobian product"""
             # evaluate the Jacobian matrix of the QNode
             variables = tfkwargs.get("variables", None)
-
             jacobian = qnode.jacobian(args, kwargs)
-            grad_output_np = grad_output.numpy()
-
-            # perform the vector-Jacobian product
-            if not grad_output_np.shape:
-                temp = grad_output_np * jacobian
-            else:
-                temp = grad_output_np.T @ jacobian
-
-            # restore the nested structure of the input args
-            grad_input = unflatten(temp.flat, args)
-
-            if isinstance(grad_input, list):
-                grad_input = [tf.convert_to_tensor(i, dtype=dtype) for i in grad_input]
-            elif isinstance(grad_input, tuple):
-                grad_input = tuple(tf.convert_to_tensor(i, dtype=dtype) for i in grad_input)
-            else:
-                grad_input = tf.convert_to_tensor(grad_input, dtype=dtype)
+            jacobian = tf.constant(jacobian, dtype=dtype)
+            grad_output_row = tf.transpose(tf.reshape(grad_output, [-1, 1]))
+            grad_input = tf.matmul(grad_output_row, jacobian)
+            grad_input = tf.reshape(grad_input, [-1])
```

Comment:
Nitpick: This is just a suggestion for code readability, feel free to modify/reject:
```suggestion
            jacobian = tf.constant(jacobian, dtype=dtype)
            
            # Reshape gradient output array as a 2D row-vector.
            grad_output_row = tf.transpose(tf.reshape(grad_output, [-1, 1]))
            
            # Calculate the vector-Jacobian matrix product, and flatten the output.
            grad_input = tf.matmul(grad_output_row, jacobian)
            grad_input = tf.reshape(grad_input, [-1])
```